server/migrations/helpers.js

```diff
@@ -1,50 +1,34 @@
 function calculatePreferences(user) {
-  const updatedPreferences = {
-    config: '',
-    sort: 0,
-    star: user?.preferences?.star || [],
-    complete: user?.preferences?.complete || [],
-  }
-  // Update complete
-  if (
-    !!user.preferences &&
-    user.preferences.complete &&
-    Object.keys(user.preferences.complete).length &&
-    !Array.isArray(user.preferences.complete)
-  ) {
-    updatedPreferences.complete = Object.keys(user.preferences.complete).map(
-      (siteName) => {
-        return {
-          site: siteName,
-          completedSubjects: user.preferences.complete[siteName],
-        }
-      }
-    )
-  }
+  const { siteSubjects } = user
+  const isUserMigrated = !!siteSubjects?.length
+  const preferences = migratePreferences(user)
 
-  //Update Starred
-  if (
-    !!user.preferences &&
-    user.preferences.star &&
-    Object.keys(user.preferences.star).length &&
-    !Array.isArray(user.preferences.star)
-  ) {
-    updatedPreferences.star = Object.keys(user.preferences.star).map(
-      (siteName) => {
-        return {
-          site: siteName,
-          starredSubjects: user.preferences.star[siteName],
-        }
-      }
-    )
+  if (isUserMigrated) {
+    return { siteSubjects, preferences }
+  } else {
+    return { siteSubjects: migrateSiteSubjects(user), preferences }
   }
+}
 
-  if (user.preferences?.config && !!user.preferences?.config)
-    updatedPreferences.config = user.preferences.config || ''
+function migrateSiteSubjects(user) {
+  const siteSubjects = []
 
-  if (user.preferences?.sort && !!user.preferences?.sort)
-    updatedPreferences.sort = user.preferences.sort || ''
-  return updatedPreferences
-}
+  user.access.forEach((site, i) => {
+    const { preferences } = user
+    const siteSubject = {
+      site,
+      starredSubjects: preferences.star[site] || [],
+      completedSubjects: preferences.complete[site] || [],
+    }
+    siteSubjects.push(siteSubject)
+  })
 
+  return siteSubjects
```

Comment:
if you map you don't need the `siteSubjects` array.

```suggestion
  return user.access.map((site) => ({
    site,
    starredSubjects: user.preferences.star[site] || [],
    completedSubjects: user.preferences.complete[site] || [],
  }))
```

Reply:
Thank you for this

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/afcceeeef4535ef07779b8ce60ba0310d80b00f8